OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -193,6 +193,8 @@ public class Aircraft : ITick, ISync, IFacing, IPositionable, IMove, IIssueOrder
 		public bool ForceLanding { get; private set; }
 		CPos? landingCell;
 
+		public WDist LandAltitude;
```

Comment:
Having activities pass state to each other via their parent traits is an ongoing source of pain.

Can we please instead pass the desired offset directly to the `HeliLand` activity?
#16365 implements this for the merged `Land` - we could either add a similar `offset` argument to `HeliLand` as a stopgap, or aim to merge that PR first and rebase this on top.

Reply:
No, passing it as an offset into `Land` is a bad idea and will break carryalls again. This might seem right for `DeliverUnit,` but the problem is that you also need this same offset whenever else `Land` is called as long as the carryall still carries the unit. Otherwise the unit will end up too low every time you make a landing (except on release). You might potentially even need to have this offset on top of another offset (e.g. carryalls landing on platforms). This really should be treated as a temporary state change in the aircrafts characteristics and should be decoupled from the activity itself.

Reply:
Ah yes, good point. What if we exposed this as methods to set/clear a custom land altitude offset, so that it is decoupled from the default value. We should then always apply this offset in Land, instead of only to the no-altitude version. Otherwise we’ll run into issues if we want to land on a docking point with a non-zero height.

Reply:
Remember that this offset would need to be applied whenever `LandAltitude` is called, for instance in `OnBecomingIdle` and `GetActorBelow`, not just the `HeliLand/Land` activities. Whenever we add a method that wants to know the landaltitude we would have to remember to add the offset. It seems to me better for maintainability if the offset is referred to by LandAltitude itself. But yeah, I could use a getter for `LandAltitude` and a separate offset parameter for that.

Reply:
I went for add/subtract methods instead of set/clear because this would allow multiple offsets to be set cumulatively, but this would still make sure `LandAltitude` is not set directly outside of `Aircraft`.